app/services/listing_index_service/data_types.rb

```diff
@@ -17,7 +17,7 @@ module ListingIndexService::DataTypes
     [:per_page, :to_integer, :mandatory, gte: 1],
     [:keywords, :string, :optional],
     [:categories, :array, :optional],
-    [:listing_shape_id, :fixnum, :optional],
+    [:listing_shape_id, :array, :optional],
```

Comment:
@ovan To my understanding sphinx can handle either a single number or an array of numbers.

@gnomet In that case you should rename it to `listing_shape_ids` in the ListingIndexService layer and then rename it back to `listing_shape_id` in SphinxAdapter. Remember that Sphinx is only an implementation detail that may be replaced with some other Adapter, e.g. ElasticSearchAdapter.